packages/http/src/payload.ts

```diff
@@ -115,7 +115,18 @@ function resolveBody(
 
   const contentTypeProperty = metadata.find((x) => x.kind === "contentType");
 
-  const file = getHttpFileModel(program, requestOrResponseType);
+  const metadataPropToMetadata = new Map<ModelProperty, HttpProperty>(
+    metadata.map((x) => [x.property, x]),
+  );
+
+  const file = getHttpFileModel(program, requestOrResponseType, (property) => {
+    const httpProperty = metadataPropToMetadata.get(property);
+    if (["filename"].includes(property.name)) return true;
```

Comment:
`filename` is the only field that made sense to allow it to have metadata.

`contents` or `contenType` having metadata causes `File` to be interpreted as just a normal model.

I _think_ that's what we expect.